src/RedemptionPool.sol

```diff
@@ -98,74 +128,88 @@ contract RedemptionPool is Ownable {
 
     /// @notice deposit GRO tokens to the pool before the deadline
     /// @param _amount amount of GRO tokens to deposit
-    function deposit(uint256 _amount) external {
-        // Checks that the DEADLINE has not passed
-        if (block.timestamp > DEADLINE) {
-            revert RedemptionErrors.DeadlineExceeded();
-        }
+    function deposit(uint256 _amount) external onlyBeforeDeadline {
+        if (_amount <= 0) revert RedemptionErrors.GreaterThanZeroOnly();
+
         // Transfers the GRO tokens from the sender to this contract
         GRO.safeTransferFrom(msg.sender, address(this), _amount);
         // Increases the balance of the sender by the amount
-        _userBalance[msg.sender] += _amount;
+        _userGROBalance[msg.sender] += _amount;
         // Increases the total deposited by the amount
         totalGRO += _amount;
         // Emits the Deposit event
-        emit Deposit(msg.sender, _amount);
+        emit Deposit(msg.sender, _amount, totalGRO);
```

Comment:
Is there a reason to lock totalGRO here?

Reply:
You mean emit? To easily see amount locked without adding up prior events? Unnecessary?

Reply:
I would say it's unnecessary, cause you can easily query blockchain using archive node and retrieve those value if it's ever needed

Reply:
Since deposit anyway emits a GRO transfer event in the GRO token... then no event needed at all actually?